### ISSUE TITLE
random 401 in web ui

### ISSUE THREAD

**Issue by sherlant** [goharbor/harbor-helm#1387]
Hi,
I try to setup harbor with external redis, pg database and s3 hosteur in ha on my k8s cluster.
I have an issue when i try to log in the web ui.
My problem is that randomly, i have 401 in request call with "UNAUTHORIZED" in body.
for example, this is a try to log in, after submit :
- POST https://harbor.my-domain.com/c/login -> 200
- GET https://harbor.my-domain.com/api/v2.0/systeminfo -> 200
- GET https://harbor.lab.cloudcrane.io/api/v2.0/users/current -> 200
- GET https://harbor.my-domain.com/api/v2.0/systeminfo -> 200
- GET https://harbor.lab.cloudcrane.io/api/v2.0/projects?with_detail=false&page_size=100&page=1 -> 200
- GET https://harbor.lab.cloudcrane.io/api/v2.0/robots?page_size=15&page=1 -> 401
- GET https://harbor.lab.cloudcrane.io/api/v2.0/export/cve/executions -> 200
- GET https://harbor.lab.cloudcrane.io/api/v2.0/users/current -> 401
When i click on login, i can see the home screen one second and i am redirect to login page.
Sometime, I can login and navigate in web ui for a quick time, but randomly, when i click on a menu link, i'm logout with same 401 errors.
And sometime, when i submit in the login page, the button stay in gray, and i must refresh manually the page to retry. 
I have show all logs, but nothing can help me.
Any idea ?
Thx

**Comment by zyyw** [comment]
How many core pod do you have? How about setting only 1 pod?
How long do you wait between a 200 request and 401? If it's a long time, maybe session timeout?

**Comment by sherlant** [comment]
I have 3 core pods. I have try to scale down to 1 all pods, same issue. I have no long time between requests, it's very quickly

**Comment by sherlant** [comment]
No more idea ?

**Comment by jkrusy** [comment]
> And sometime, when i submit in the login page, the button stay in gray, and i must refresh manually the page to retry.

I get this too on a fresh installation via helm.
`/api/v2.0/users/current` always gets 
```json
{"errors":[{"code":"UNAUTHORIZED","message":"unauthorized"}]}
```
on login

**Comment by thavlik** [comment]
I am experiencing this as well and not able to login.

**Comment by neelgriddalur** [comment]
Any update on this issue within the community? Been facing this as well.

**Comment by ateijelo** [comment]
For what it's worth, besides the web portal, I also got random `401 Unexpected EOF` when doing all other kind of tasks with the registry. Pulling and pushing images, pulling charts, etc. And after a lot of troubleshooting, I noticed random disconnections between harbor-core and my external PostgreSQL database.

I tried disabling idle connections completely by setting:

```yaml
database:
  type: external
  maxIdleConns: 0  # don't keep idle connections
```

I imagine this will have a negative impact on performance, but at least the 401s seem to have disappeared for good.
